chirp/drivers/tdh8.py

```diff
@@ -0,0 +1,2193 @@
+                                list_val.append(0xFF)
+                                lenth_val += 1
+                        self._memobj.group4.group4 = list_val
+
+                    elif setting == 'group5' and element.value.get_mutable():
+                        val = str(element.value)
+                        list_val = []
+                        lenth_val = 0
+                        while lenth_val < (len(val)):
+                            if val[lenth_val] != ' ':
+                                list_val.append(int(val[lenth_val], 16))
+                                lenth_val += 1
+                            else:
+                                list_val.append(0xFF)
+                                lenth_val += 1
+                        self._memobj.group5.group5 = list_val
+
+                    elif setting == 'group6' and element.value.get_mutable():
+                        val = str(element.value)
+                        list_val = []
+                        lenth_val = 0
+                        while lenth_val < (len(val)):
+                            if val[lenth_val] != ' ':
+                                list_val.append(int(val[lenth_val], 16))
+                                lenth_val += 1
+                            else:
+                                list_val.append(0xFF)
+                                lenth_val += 1
+                        self._memobj.group6.group6 = list_val
+
+                    elif setting == 'group7' and element.value.get_mutable():
+                        val = str(element.value)
+                        list_val = []
+                        lenth_val = 0
+                        while lenth_val < (len(val)):
+                            if val[lenth_val] != ' ':
+                                list_val.append(int(val[lenth_val], 16))
+                                lenth_val += 1
+                            else:
+                                list_val.append(0xFF)
+                                lenth_val += 1
+                        self._memobj.group7.group7 = list_val
+
+                    elif setting == 'group8' and element.value.get_mutable():
+                        val = str(element.value)
+                        list_val = []
+                        lenth_val = 0
+                        while lenth_val < (len(val)):
+                            if val[lenth_val] != ' ':
+                                list_val.append(int(val[lenth_val], 16))
+                                lenth_val += 1
+                            else:
+                                list_val.append(0xFF)
+                                lenth_val += 1
+                        self._memobj.group8.group8 = list_val
+                    elif element.value.get_mutable():
+                        setattr(obj, setting, element.value)
+                except Exception as e:
+                    LOG.debug(element.get_name())
+                    raise
+
```

Comment:
This whole `set_settings()` method is reeeeally long. It might be nice to break it up in the future, like you have already done in some cases, but I won't make you do it now.